plugins/modules/asa_config.py

```diff
@@ -303,9 +333,23 @@ def run(module, result):
         result["changed"] = True
 
     if module.params["save"]:
-        if not module.check_mode:
-            run_commands(module, "write mem")
-        result["changed"] = True
+        module.warn(
+            "module param save is deprecated, please use newer and updated param save_when instead which is released with more functionality!"
+        )
+        save_config(module, result)
+    if module.params["save_when"] == "always":
```

Comment:
The `if..elif` block should be checked for if `module.check_mode` is enabled or not as currently if `check_mode` is enabled still it will edit the actual database on the device which is not expected.